plugins/SolidView/SolidView.py

```diff
@@ -175,4 +260,49 @@ def beginRendering(self):
                     renderer.queueNode(scene.getRoot(), mesh = node.getBoundingBoxMesh(), mode = RenderBatch.RenderMode.LineLoop)
 
     def endRendering(self):
-        pass
+        # check whether the xray overlay is showing badness
+        if time.time() > self._next_xray_checking_time\
+                and CuraApplication.getInstance().getPreferences().getValue(self._show_xray_warning_preference):
+            self._next_xray_checking_time = time.time() + self._xray_checking_update_time
+
+            xray_img = self._xray_pass.getOutput()
+            xray_img = xray_img.convertToFormat(QImage.Format.Format_RGB888)
+            ptr = xray_img.bits()
+            ptr.setsize(xray_img.byteCount())
+            reds = np.array(ptr).reshape(xray_img.height(), xray_img.width(), 3)[:,:,0]  # Copies the data
+            bad_pixel_count = np.sum(np.mod(reds, 2)) # check number of pixels with an odd intersection count
+
+            if bad_pixel_count > 10: # allow for 10 pixels to be erroneously marked as problematic
```

Comment:
I thought it might be possible that there are ever so tiny creases or perhaps there are some rounding errors in the GPU which might cause false negatives.

I'm okay with this being set to a different hard coded value...

Reply:
Leaving this as-is for now until testing.

Reply:
I think that you originally had to set a limit but not due to rounding errors. You were taking data directly from the pixel buffer of the image, but his pixel buffer has padding bytes to align each column of pixels to word sizes. These padding bytes are not initialised, so they contain random noise. These could be recognised as bad pixels. When I fixed that, the noise was completely gone. Normally the rounding errors in triangle coordinates don't cause gaps to appear between the triangles; otherwise you'd see gap pixels in every 3D model.